search-service/src/main/kotlin/com/egm/stellio/search/service/EntityEventListenerService.kt

```diff
@@ -127,4 +142,51 @@ class EntityEventListenerService(
             logger.debug("Created new attribute instance for temporal entity attribute (${it.t1})")
         }.subscribe()
     }
+
+    fun handleAttributeAppend(
+        entityId: URI,
+        attributeName: String,
+        attributeValuesNode: JsonNode,
+        updatedEntity: String,
+        contexts: List<String>
+    ) {
+        if (!attributeValuesNode.has("observedAt")) return
+
+        val expandedAttributeName = expandJsonLdKey(attributeName, contexts)!!
+        val rawAttributeValue = attributeValuesNode["value"]
+        val parsedAttributeValue =
+            if (rawAttributeValue.isNumber)
+                Pair(null, valueToDoubleOrNull(rawAttributeValue.asDouble()))
+            else
+                Pair(valueToStringOrNull(rawAttributeValue.asText()), null)
+
+        val attributeValueType =
+            if (parsedAttributeValue.second != null) TemporalEntityAttribute.AttributeValueType.MEASURE
+            else TemporalEntityAttribute.AttributeValueType.ANY
+
+        val temporalEntityAttribute = TemporalEntityAttribute(
+            entityId = entityId,
+            type = expandJsonLdEntity(updatedEntity).type,
+            attributeName = expandedAttributeName,
+            attributeValueType = attributeValueType,
+            datasetId = attributeValuesNode["datasetId"]?.asText()?.toUri()
+        )
+        val attributeInstance = AttributeInstance(
+            temporalEntityAttribute = temporalEntityAttribute.id,
+            observedAt = ZonedDateTime.parse(attributeValuesNode["observedAt"].asText()),
+            measuredValue = parsedAttributeValue.second,
+            value = parsedAttributeValue.first
+        )
+
+        temporalEntityAttributeService.create(temporalEntityAttribute).zipWhen {
+            attributeInstanceService.create(attributeInstance).then(
+                temporalEntityAttributeService.updateEntityPayload(entityId, updatedEntity)
+            )
+        }
+            .doOnError {
+                logger.error("Failed to persist new temporal entity attribute, ignoring it", it)
+            }.doOnNext {
+                logger.debug("Created new temporal entity attribute with one attribute instance")
+            }.subscribe()
```

Comment:
We have a very similar business logic in `TemporalEntityAttributeService`, no easy way to mutualize code?

Reply:
Just checked, the attributeValueType is not created the same way.
Also, in createEntityTemporalReferences we are not doing exactly the same thing since we are creating the EntityPayload but here we are calling the updateEntityPayload function.

Reply:
OK, let's keep it for a refactoring session later